src/Microsoft.AspNet.Server.Kestrel/Http/FrameRequestStream.cs

```diff
@@ -94,7 +94,7 @@ public Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationTok
                 {
                     tcs2.SetResult(task2.Result);
                 }
-            }, tcs);
+            }, tcs, cancellationToken);
             return tcs.Task;
```

Comment:
Is kinda internal; hence why the suggestion; before anyone takes a dependency on it...